### ISSUE TITLE
max char limit on short description with unicode fails

### ISSUE THREAD

**Issue by alexrashed** [peter-evans/dockerhub-description#142]
I just stumbled across a weird issue, where the action would suddenly fail (after quite a while not being executed at all).
After digging into the issue, it seems that the `short-description` limit is not correctly enforced.
The `short-description` input was introduced with https://github.com/peter-evans/dockerhub-description/commit/8fa820167718d4572cf4fb7bd1b361365a8b7708, which already truncates inputs longer than 100 characters (since otherwise the API would return a status code 400 / Bad Request):
https://github.com/peter-evans/dockerhub-description/blob/3ba533f0ffa0cf9780bd72010bc80d0cac3a805b/src/dockerhub-helper.ts#L39
However, it seems that the limit differs when using supplementary characters like emojis. I did some tests, and it turns out that 1 supplementary character seems to be equal to 4 ASCII characters:
- 25 emojis, successful: https://github.com/alexrashed/dockerhub-description-unicode/actions/runs/4617334701/jobs/8163468328
- 100 ASCII characters, successful: https://github.com/alexrashed/dockerhub-description-unicode/actions/runs/4617378471/jobs/8163566247
- 26 emojis, 400 / Bad Request: https://github.com/alexrashed/dockerhub-description-unicode/actions/runs/4617345906

The run with 26 emojis results in the following error:
```
##[debug]Inputs: {
##[debug]  username: '***',
##[debug]  password: '***',
##[debug]  repository: '***/dockerhub-description-tests',
##[debug]  shortDescription: '💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻💻',
##[debug]  readmeFilepath: './README.md',
##[debug]  enableUrlCompletion: false,
##[debug]  imageExtensions: 'bmp,gif,jpg,jpeg,png,svg,webp'
##[debug]}
Reading description source file
##[debug]💻
Acquiring token
::add-mask::***
Sending PATCH request
##[debug]Error: Bad Request
##[debug]    at /home/runner/work/_actions/peter-evans/dockerhub-description/v3/dist/index.js:83:23
##[debug]    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: Bad Request
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Docker Hub Description
```

This could be explained by UTF-8 being a variable-length content-encoding where code-points with lower numerical values are using fewer bytes. This would basically mean that the limit of 100 is actually a limit of 100 bytes, not 100 characters.

**Comment by peter-evans** [comment]
Hi @alexrashed 

Thank you for your detailed report of this issue. It would not surprise me that it should be 100 bytes, not 100 characters. The readme content already works this way and I [truncate to a byte limit](https://github.com/peter-evans/dockerhub-description/blob/3ba533f0ffa0cf9780bd72010bc80d0cac3a805b/src/readme-helper.ts#L31-L37) while respecting unicode surrogate pairs. I just need to add that functionality to the `short-description`, too, it seems. I'll make a fix for this soon.

**Comment by peter-evans** [comment]
Released a fix to truncate the short description in `v3.4.0` / `v3`.

Thank you!
